web/ui/src/Services/AppsController.js

```diff
@@ -292,34 +267,108 @@
                 $scope.templates.data = utils.mapToArr(templates);
             });
         }
-       
-        function fetchInternalServiceInstances(app) {
-            return resourcesFactory.v2.getInternalServiceInstances(app.id).then( data => {
-                app.instances = data;
+
+        function refreshApps() {
+            return fetchApps()
+                .then(fillOutApps)
+                .then(fetchServicesHealth)
+                .then(fetchInternalServicesHealth)
+                .then(() => touch());
```

Comment:
need a `catch` in here or this beautiful promise chain will be the source of infinite sadness.